test/action_controller/serialization_test.rb

```diff
@@ -265,7 +264,7 @@ def test_render_with_cache_enable
         assert_equal expected.to_json, @response.body
 
         get :render_changed_object_with_cache_enabled
-        assert_equal expected.to_json, @response.body
+        assert_not_equal expected.to_json, @response.body
```

Comment:
This used to be able to be `assert_equal` because the cache key wasn't a high enough resolution to detect the change. This was actually incorrect (and/or bad caching), so at a minimum I changed the test to assert on the correct result, which it now receives.